HW1/P2/P2b.py

```diff
@@ -1,3 +1,27 @@
 from P2 import *
 
 # Your code here
+from pyspark import SparkContext
+import matplotlib.pyplot as plt
+import numpy as np
+
+sc = SparkContext("local[4]", "HW1-2b Default Partioning")
+sc.setLogLevel("ERROR")
+
+#set up 2000x2000 array and transform array into [-2,2]x[-2,2] grid... 
+rdd = sc.parallelize(xrange(0,2001),10)
+grid = rdd.cartesian(rdd).repartition(100) #Creates ordered pairs, redistribute randomly...
```

Comment:
@jdlkem I was also surprise that .repartition( ) didn't do a better job at normalizing the compute across partitions. I tried to do .partitionBy( ) initially but couldn't figure out how to use hash functions properly.